apps/frontend/src/modules/surveys/components/studio/create/editor-panel/components/EditorTextInput.tsx

```diff
@@ -21,89 +25,137 @@ export const EditorTextInput = ({
     className,
     showCharacterCount = true,
     maxCharacterCount = 500,
-    placeholder,
     defaultValue,
+    options,
     onChange,
-    ...props
 }: EditorTextProps) => {
-    const [atBtnClicked, setAtBtnClicked] = useState(false);
-    const [atIndex, setAtIndex] = useState<number | null>(null);
-    const [selectedOption, setSelectedOption] = useState<Option | null>(null);
-    const [inputValue, setInputValue] = useState(defaultValue || "");
+    const [textContent, setTextContent] = useState(decode(defaultValue ?? ""));
+
+    const [displayFallbackField, setDisplayFallbackField] = useState(false);
+
+    const modules: StringMap = useMemo(() => {
```

Comment:
@afolasope You can also move `modules` out of this component since it doesn't use component state or cause any side effect.

Reply:
okay

Reply:
@afolasope Did you use `useMemo` where you moved it?

Reply:
No